pkg/types/ccipocr3/interfaces.go

```diff
@@ -50,4 +51,5 @@ type TokenDataEncoder interface {
 type EstimateProvider interface {
 	CalculateMerkleTreeGas(numRequests int) uint64
 	CalculateMessageMaxGas(msg Message) uint64
+	CalculateUsdPerUnitGas(sourceGasPrice *big.Int, usdPerFeeCoin *big.Int) *big.Int
```

Comment:
Actually I don't think this interface would work, you need to have a source chain selector argument somewhere